Checkout Block: Show load indicators for shipping options only when relevant address fields change (#57374)
woocommerce/woocommerce: plugins/woocommerce/client/blocks/docs/third-party-developers/extensibility/hooks/filters.md

---

## woocommerce_address_fields_for_shipping_rates

Filters the shipping address fields that trigger shipping rate recalculation.

```php
apply_filters( 'woocommerce_address_fields_for_shipping_rates', array $fields )
```

### Description

> By default, WooCommerce considers the following shipping fields as essential for shipping rate calculations: `'state'`, `'country'`, `'postcode'`, and `'city'`.

Reviewer: ```suggestion
By default, Store API considers the following shipping fields as essential for shipping rate calculations: `state`, `country`, `postcode`, and `city`.
```

Personally I think it looks better without the quotes, but don't feel that strongly about it if you'd prefer to skip. I do think we need to specify that it's only Store API that does this though, shortcode only does city and postcode.

Author: Great suggestion regarding Store API. I'll update the paragraph.

This filter allows developers to add new fields to the list above.

> The filter is used in the checkout blocks powered checkout process. 

Reviewer: ```suggestion
The filter is used in the blocks-powered checkout process. 
```

### Parameters

| Argument | Type  | Description                                                                                                                                                                           |
| -------- | ----- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| $fields  | array | An array of address field names (strings) that should trigger shipping rate recalculation. Defaults to an empty array, which means only the core fields are used unless you add more. |

### Returns

`array` The updated list of address field names that should trigger shipping rate recalculation.

### Example

```php
add_filter( 'woocommerce_address_fields_for_shipping_rates', function( $fields ) {
    // Add a custom field to the list.

>     return ['address_1'];

Reviewer: ```suggestion
    return [ 'address_1' ];
```
